torch/optim/adagrad.py

```diff
@@ -64,6 +76,13 @@ def step(self, closure=None):
                 grad = p.grad.data
                 state = self.state[p]
 
+                # state initialization
+                if len(state) == 0:
+                    state['step'] = 0
+                    state['sum'] = torch.full_like(p.data, group['initial_accumulator_value'])
+                    if self._is_share_memory:
+                        state['sum'].share_memory_()
```

Comment:
cc @VitalyFedyunin If you ever make `share_memory ` a keyword argument like `pin_memory`, this is a good site to kill :)

Reply:
Only after we move all functions from the python storage to the aten's Tensors.